src/gui/page.py

```diff
@@ -0,0 +1,141 @@
+                                                text=label,
+                                                id=button_id,
+                                                is_disabled=True))
+                current_height += self._buttons[-1].get_height() + Page.VERTICAL_SPACE
+            elif button_type == ButtonType.MULTI_CHOICE_SHORT:
+                self._buttons.append(MultiChoiceButton(pos_x=Page.MARGIN_LEFT,
+                                                       pos_y=current_height,
+                                                       width=Page.USABLE_AREA_WIDTH,
+                                                       height=Page.BUTTON_HEIGHT,
+                                                       choices=label,
+                                                       id=button_id))
+                current_height += self._buttons[-1].get_height() + Page.VERTICAL_SPACE
+            elif button_type == ButtonType.MULTI_CHOICE_LONG:
+                self._buttons.append(MultiChoiceButton(pos_x=Page.MARGIN_LEFT,
+                                                       pos_y=current_height,
+                                                       width=Page.USABLE_AREA_WIDTH,
+                                                       height=Page.BUTTON_HEIGHT * 3 + Page.VERTICAL_SPACE * 2,
+                                                       choices=label,
+                                                       id=button_id))
+                current_height += self._buttons[-1].get_height() + Page.VERTICAL_SPACE
+
+        for i, button in enumerate(self._buttons):
+            if not button.is_disabled():
+                self._active_button_idx = self._default_active_button_idx = i
+                break
+
+    def move_up(self):
+        if self._active_button_idx > 0:
+            next_idx = self._active_button_idx - 1
+            while self._buttons[next_idx].is_disabled():
+                next_idx -= 1
+                if next_idx < 0:
+                    return
+            self._active_button_idx = next_idx
+            while self._active_button_idx > 0 and self._buttons[self._active_button_idx].get_width() == 115:
+                self._active_button_idx -= 1
+
+    def move_down(self):
+        if self._active_button_idx < len(self._buttons) - 1:
+            next_idx = self._active_button_idx + 1
+            while self._buttons[next_idx].is_disabled():
+                next_idx += 1
+                if next_idx > len(self._buttons) - 1:
+                    return
+            self._active_button_idx = next_idx
+
+    def move_left(self):
+        if type(self._buttons[self._active_button_idx]) is MultiChoiceButton:
+            self._buttons[self._active_button_idx].go_left()
+        else:
+            prev_idx = self._active_button_idx - 1
+            if prev_idx >= 0 and self._buttons[prev_idx].get_width() == Page.narrow_button_size:
```

Comment:
I would somehow differently differ the type of button. You can create two classes or make a method that makes the check if width is equal to narrow button

Reply:
But if I create two classes then I have twice much the same code. And about the function - I don't like if they have just one line.